lib/rspec/core/metadata_filter.rb

```diff
@@ -61,6 +58,14 @@ def line_number_filter_applies?(line_numbers, metadata)
           !(relevant_line_numbers(metadata) & preceding_declaration_lines).empty?
         end
 
+        def proc_filter_applies?(key, proc, metadata)
+          case proc.arity
+          when 0 then proc.call
+          when 2 then proc.call(metadata[key], metadata)
+          else proc.call(metadata[key])
+          end
+        end
```

Comment:
This extraction is a good refactoring, but is unrelated to the rest of the changes in this commit.  Do you mind putting it in its own commit?